packages/jazz-tools/src/tests/deepLoading.test.ts

```diff
@@ -610,6 +610,34 @@ describe("Deep loading with unauthorized account", async () => {
 
     errorSpy.mockReset();
   });
+
+  test("setting null via proxy", async () => {
+    class Lv1 extends CoMap {
+      lv2 = co.ref(Lv2);
+    }
+
+    class Lv2 extends CoMap {
+      lv3 = co.optional.ref(Lv3);
+    }
+
+    class Lv3 extends CoMap {
+      string = co.string;
+    }
+
+    const map = Lv1.create(
+      { lv2: Lv2.create({ lv3: Lv3.create({ string: "hello" }, bob) }, bob) },
+      bob,
+    );
+
+    map.lv2!.lv3 = null;
```

Comment:
Found that when the property is written via setter the value becomes `null`

Instead when an optional value is set as null in the constructor is casted to `undefined`